rules/valid-expect.js

```diff
@@ -5,19 +5,72 @@
  * MIT license, Tom Vincent.
  */
 
-const getDocsUrl = require('./util').getDocsUrl;
+const get = require('lodash.get');
+const util = require('./util');
 
 const expectProperties = ['not', 'resolves', 'rejects'];
 
 module.exports = {
   meta: {
     docs: {
-      url: getDocsUrl(__filename),
+      url: util.getDocsUrl(__filename),
     },
   },
   create(context) {
+    function validateAsyncExpects(node) {
+      const callback = node.arguments[1];
+      if (
+        callback &&
+        util.isFunction(callback) &&
+        callback.body.type === 'BlockStatement'
+      ) {
+        callback.body.body
+          .filter(node => node.type === 'ExpressionStatement')
+          .filter(node => {
+            const objectName = get(
+              node,
+              'expression.callee.object.object.callee.name'
+            );
+            const propertyName = get(
+              node,
+              'expression.callee.object.property.name'
+            );
+
+            return (
+              node.expression.type === 'CallExpression' &&
+              objectName === 'expect' &&
+              (propertyName === 'resolves' || propertyName === 'rejects')
```

Comment:
I wonder if we can reuse the logic somehow with the other `resolves`/`rejects` checks

Reply:
Good idea. I'll look into that while resolving the failing test case you reported below.